lib/arbitraryjson.js

```diff
@@ -0,0 +1,38 @@
+const ops = require('./ops')
+
+module.exports = defaults
+
+function defaults (opts) {
+  const { ch, code } = ops(opts)
+  const buffer = !!opts.buffer
+
+  return compileArbitraryJSON
+
+  function compileArbitraryJSON (gen, prop, rawSchema, compileAny) {
+    const {indexOf, stringSlice} = ops({buffer})
+
+    gen(`
+      let arbjson = 0
+      let startjson = ptr
+      do {
+        if (${ch('ptr')} === ${code('"')}) {
+          var i = ${indexOf('"', '++ptr')}
+
+          while (${ch('i - 1')} === ${code('\\')}) {
+            var cnt = 1
+            while (${ch('i - 1 - cnt')} === ${code('\\')}) cnt++
+            if ((cnt & 1) === 0) break
+            i = ${indexOf('"', 'i + 1')}
+          }
+          ptr = i
+        }
+        else if (${ch('ptr')} === ${code('{')}) ++arbjson
+        else if (${ch('ptr')} === ${code('}')}) --arbjson
```

Comment:
i think you need to handle arrays here also, ie [ and ]

Reply:
The idea is that we just skip through the data as fast as possible to find the end of the object. Then pass the data to json.parse. We need to handle strings because they can contain { and thus trip our "where is the end".